### PR TITLE
[WIP] Add and use isEmpty() public method to the BlobPath.java

### PR DESCRIPTION
**Pull request by alexVengrovsk**

### CODE REVIEW

**Comment by alexVengrovsk** [comment]
I signed the CLA as Alex Vengrovskyi and I am a member of Elasticsearch

File: core/src/main/java/org/elasticsearch/common/blobstore/url/URLBlobStore.java [hunk 1]
```diff
@@ -119,10 +119,10 @@ public void close() {
      * @return Base URL + path
      */
     private URL buildPath(BlobPath path) throws MalformedURLException {
-        String[] paths = path.toArray();
-        if (paths.length == 0) {
+        if (path.isEmpty) {
```
Comment (dadoonet) [inline core/src/main/java/org/elasticsearch/common/blobstore/url/URLBlobStore.java]: Should it be `isEmpty()` here? Wondering how this is compiling but I might be wrong.

Comment (alexVengrovsk) [inline core/src/main/java/org/elasticsearch/common/blobstore/url/URLBlobStore.java]: @dadoonet I added isEmpty() public method to BlobPath class in first [commit](https://github.com/alexVengrovsk/elasticsearch/commit/6898f877e0b0c43bdddd600cbc0c1bdf88df85cd) of this PR

Comment (alexVengrovsk) [inline core/src/main/java/org/elasticsearch/common/blobstore/url/URLBlobStore.java]: It should be ok

Comment (dadoonet) [inline core/src/main/java/org/elasticsearch/common/blobstore/url/URLBlobStore.java]: I saw it. Just wondering how `path.isEmpty` works as I would expect `path.isEmpty()`.

**Comment by alexVengrovsk** [comment]
@dadoonet I am sorry for such a mistake. I fixed it. Thank you!

File: core/src/main/java/org/elasticsearch/common/blobstore/url/URLBlobStore.java (hunk 1 above)
Comment (alexVengrovsk) [inline core/src/main/java/org/elasticsearch/common/blobstore/url/URLBlobStore.java]: I fixed it. Thank you!

**Comment by dadoonet** [comment]
@alexVengrovsk So you did not compile your code before sending the PR, right?
BTW I'm curious if you need this for a repository plugin or something else?

@imotov Wanna review this change?

**Comment by alexVengrovsk** [comment]
Yes, I didn't. I made changes using github.

**Comment by dadoonet** [comment]
@alexVengrovsk So can you please explain why you need this change?
And if you want it to be merged, could you please run `gradle test` so we are all sure you did not break anything.

**Comment by alexVengrovsk** [comment]
I can do it later. I wonted to make an improvemeth to the code, to make a contribution to the project.

**Comment by dadoonet** [comment]
Thank you. We really appreciate contributions in code, tests, documentation, wherever!

**Comment by clintongormley** [comment]
@alexVengrovsk I think David is asking why this code should be added? What's the use case? Also, any changes should be accompanied by tests and (if required), documentation.

**Comment by dadoonet** [comment]
No news on that and does not seem really needed. Closing.

**Pull request closed without merging**